packages/oid4vci-demo-frontend/src/utils/ObjectUtils.ts

```diff
@@ -7,7 +7,7 @@ export const extractKeys = (data: DataFormRow[]): string[] => {
 
 export const extractRequiredKeys = (data: DataFormRow[]): string[] => {
   return data.flatMap(item =>
-      item.filter(field => field.required)
+      item.filter(field => field.required && !Boolean(field.leaveEmptyInCaseOfEmptyValue))
```

Comment:
why are we wrapping a boolean in a boolean?

Reply:
to handle optional part better. otherwise I have to check for it being undefined or false

Reply:
i don't see why !field.leaveEmptyInCaseOfEmptyValue should not work. can you test that?

Reply:
changed it to your suggestion

Reply:
you do not state the result, so i am going to assume it worked